strongbox-event-api/src/main/java/org/carlspring/strongbox/event/EventExecutorFactoryBean.java

```diff
@@ -0,0 +1,130 @@
+package org.carlspring.strongbox.event;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+
+import javax.servlet.ServletContext;
+
+import org.apache.catalina.Service;
+import org.apache.catalina.core.ApplicationContext;
+import org.apache.catalina.core.ApplicationContextFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.core.task.SyncTaskExecutor;
+import org.springframework.data.util.ReflectionUtils;
+
+public class EventExecutorFactoryBean implements FactoryBean<Executor>
```

Comment:
why can't we use default `org.springframework.core.task.SimpleAsyncTaskExecutor` or even custom `java.util.concurrent.ThreadPoolExecutor` ? I don't like it it's container specific solution

Reply:
the thing is that if we produce Async events within request thread and then execute them using another executor this can cause Event queue overflowing. 

I have tried it with limit of 100 Events in queue and this was not enought even for our strongbox-against-strongbox. In real production life there will be much more events. 


After understanding of this problem I came with this solution to share same executor for HTTP Requests and Async Events, which will bring correlation for events and requests and we will not be able to produce more events then we can execute.

Reply:
I would try `java.util.concurrent.Executors#newFixedThreadPool(int)`

Reply:
I have tried separate executor and there was problems with event queue overflow, see my comment above.